benchmark/benchmarks.jl

```diff
@@ -23,7 +33,22 @@ const params = Dict(
     ),
 )
 
-SUITE["default"] = @benchmarkable ParticleDA.run_particle_filter($(Model.init), $(params), $(BootstrapFilter())) seconds=30 setup=(cd(mktempdir()))
+const nprt_per_rank = Int(params["filter"]["nprt"] / my_size)
+const model_data = Model.init(params["model"]["llw2d"], nprt_per_rank, my_rank)
+const bootstrap_filter_data = ParticleDA.init_filter(ParticleDA.get_params(ParticleDA.FilterParameters, params["filter"]), model_data, nprt_per_rank, Float64)
+const filter_params = ParticleDA.get_params(ParticleDA.FilterParameters, params["filter"])
+
+SUITE["base"]["get_particles"] = @benchmarkable ParticleDA.get_particles($(model_data))
+SUITE["base"]["get_mean_and_var!"] = @benchmarkable ParticleDA.get_mean_and_var!(statistics, $(ParticleDA.get_particles(model_data)), $(filter_params.master_rank)) setup=(statistics=similar(bootstrap_filter_data.statistics))
+SUITE["base"]["update_truth!"] = @benchmarkable ParticleDA.update_truth!($(model_data), $(nprt_per_rank))
+SUITE["base"]["update_particle_noise!"] = @benchmarkable ParticleDA.update_particle_noise!($(model_data), $(nprt_per_rank))
+SUITE["base"]["get_log_weights!"] = @benchmarkable ParticleDA.get_log_weights!(weights, truth_observations, model_observations, $(filter_params.weight_std)) setup=(weights = Vector{Float64}(undef, nprt_per_rank); truth_observations=ParticleDA.update_truth!(model_data, nprt_per_rank); model_observations = ParticleDA.get_particle_observations!(model_data, nprt_per_rank))
+SUITE["base"]["normalized_exp!"] = @benchmarkable ParticleDA.normalized_exp!(weights) setup=(weights = rand(filter_params.nprt))
+SUITE["base"]["resample!"] = @benchmarkable ParticleDA.resample!(resampling_indices, weights) setup=(resampling_indices = Vector{Int}(undef, filter_params.nprt); weights = rand(filter_params.nprt))
+# SUITE["base"]["copy_states!"] = @benchmarkable ParticleDA.copy_states!($(ParticleDA.get_particles(model_data)), $(bootstrap_filter_data.copy_buffer), $(bootstrap_filter_data.resampling_indices), $(my_rank), $(nprt_per_rank))
```

Comment:
This is failing inside `MPI.Irecv!` but I don't understand why.  Not sure it's worth spending more time on this, but I left it commented in case someone wants to get back to this in the future.